src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -83,6 +98,10 @@ public CSharpType(TypeProvider implementation, string ns, string name, bool isVa
         public string Name { get; }
         public bool IsValueType { get; }
         public bool IsEnum { get; }
+        public bool IsLiteral { get; set; }
+        public object? LiteralValue { get; set; }
```

Comment:
I think we should probably use `Constant` here?  https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Shared/Constant.cs